tests/components/sharkiq/test_shark_iq.py

```diff
@@ -225,29 +218,20 @@ async def test_coordinator_match(hass: HomeAssistant):
 
     coordinator = SharkIqUpdateCoordinator(hass, None, ayla_api, [shark_vac1])
 
-    # The first should succeed, the second should fail
-    api1 = SharkVacuumEntity(shark_vac1, coordinator)
-    try:
-        _ = SharkVacuumEntity(shark_vac2, coordinator)
-    except RuntimeError:
-        api2_failed = True
-    else:
-        api2_failed = False
-    assert api2_failed
-
+    api = SharkVacuumEntity(shark_vac1, coordinator)
```

Comment:
Will you refactor the tests in a separate PR?

Reply:
That's the plan.  I'm a little pressed for time right now, but wanted to knock out the core functionality issues.

Reply:
Good!